src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,483 @@
+
+    SUB_ZERO_MINING("SubZero Mining",
+        100,
+        { currentLevel -> floor((currentLevel + 1.0).pow(2.3)) },
+        { level -> mapOf(HotmReward.MINING_FORTUNE to 1.0 * level) }),
+    SURVEYOR("Surveyor",
+        20,
+        { currentLevel -> (currentLevel + 1.0).pow(4) },
+        { level -> mapOf(HotmReward.UNKNOWN to 0.75 * level) }),
+    EAGER_ADVENTURER("Eager Adventurer",
+        100,
+        { currentLevel -> floor((currentLevel + 1.0).pow(2.3)) },
+        { level -> mapOf(HotmReward.MINING_SPEED to 2.0 * level) }),
+
+    DEAD_MANS_CHEST("Dead Man's Chest",
+        50,
+        { currentLevel -> floor((currentLevel + 1.0).pow(3.2)) },
+        { level -> mapOf(HotmReward.UNKNOWN to 1.0 * level) }),
+
+    GIFTS_FROM_THE_DEPARTED("Gifts from the Departed",
+        100,
+        { currentLevel -> floor((currentLevel + 1.0).pow(2.45)) },
+        { level -> mapOf(HotmReward.UNKNOWN to 0.2 * level) }),
+
+    EXCAVATOR(
+        "Excavator",
+        50,
+        { currentLevel -> (currentLevel + 1.0).pow(3) },
+        { level -> mapOf(HotmReward.UNKNOWN to 0.5 * level) }),
+    RAGS_TO_RICHES("Rags to Riches",
+        50,
+        { currentLevel -> floor((currentLevel + 1.0).pow(3.05)) },
+        { level -> mapOf(HotmReward.MINING_FORTUNE to 2.0 * level) }),
+
+    KEEN_EYE("Keen Eye", 1, { null }, { emptyMap() }), MINESHAFT_MAYHEM(
+        "Mineshaft Mayhem",
+        1,
+        { null },
+        { emptyMap() }),
+    FROZEN_SOLID(
+        "Frozen Solid",
+        1,
+        { null },
+        { emptyMap() }),
+    GEMSTONE_INFUSION("Gemstone Infusion", 1, { null }, { emptyMap() }), HAZARDOUS_MINER(
+        "Hazardous Miner",
+        1,
+        { null },
+        { emptyMap() }), ;
+
+    private val guiNamePattern by repoGroup.pattern("perk.name.${name.lowercase().replace("_", "")}", "§.$guiName")
+
+    val rawLevel: Int
+        get() = storage?.perks?.get(this.name)?.level ?: 0
+
+    var activeLevel: Int
+        get() = storage?.perks?.get(this.name)?.level?.plus(if (HotmAPI.isBlueEggActive) 1 else 0) ?: 0
+        private set(value) {
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.level = value
+        }
+
+    var enabled: Boolean
+        get() = storage?.perks?.get(this.name)?.enabled ?: false
+        private set(value) {
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.enabled = value
+        }
+
+    var isUnlocked: Boolean
+        get() = storage?.perks?.get(this.name)?.isUnlocked ?: false
+        private set(value) {
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.isUnlocked = value
+        }
+
+    var slot: Slot? = null
+        private set
+
+    fun getLevelUpCost() = costFun(rawLevel)
+
+    fun getReward() = rewardFun(activeLevel)
+
+    fun calculateTotalCost(desiredLevel: Int) = (0 until desiredLevel).sumOf { level -> costFun(level) ?: 0.0 }.toInt()
```

Comment:
Shouldn't it start with 1 not 0?
The error is only quiet small about 10-100 powder for most perks, but still it should be correct. Use the calculator in fandom for one perk to verify the cost.
Also there would be a problem if desired level is 1, since this must always be 0 which it isn't in the way you call it here.

Reply:
I tried it with both a 0 and a 1, both numbers didnt match the data from the wiki.
Problem with the 1 is, as you stated, that it will no longer work.
What should I do then?

Reply:
Ok, will have a look. Because the need to match up. So there is a problem in the data.

Reply:
```1 until disiredLevel``` works. So simply change the 0 to 1.
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/2a983e73-29eb-422a-a9cb-43fe331882b2)
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/29cda42c-f894-4350-8976-14e1617ec8f2)